### PR TITLE
fix: replace expiring signed URLs with stable attachment:// scheme

### LINKED ISSUE #147: Added images don't show in iOS app
screenshot attached. it's worth checking if it works properly on other platforms.

Image

### PR DESCRIPTION
**Pull request by JakubAnderwald**
## Summary

- Introduces an `attachment://` URL scheme to store stable file paths in note content instead of expiring Supabase signed URLs (7-day expiry)
- Resolves `attachment://` URLs to fresh signed URLs at render time on both web (server-side in GET handler) and mobile (before passing content to TipTap)
- Lazily migrates legacy signed URLs to `attachment://` on save (both platforms), ensuring backward compatibility

## Problem

Images added via the web editor stored Supabase signed URLs directly in note content JSON. These URLs expire after 7 days, permanently breaking images in both web and mobile apps.

## Changes

| File | Change |
|------|--------|
| `packages/shared/src/editor/attachment-url.ts` | **New** — URL scheme helpers (`toAttachmentUrl`, `isAttachmentUrl`, `extractFilePath`, signed URL detection) |
| `packages/shared/src/editor/resolve-urls.ts` | **New** — Content tree walkers for resolving/migrating URLs in BlockNote and TipTap formats |
| `packages/shared/src/constants.ts` | Add `ATTACHMENT_URL_PREFIX` constant |
| `packages/shared/src/index.ts` | Export new functions |
| `apps/web/src/components/editor/note-editor.tsx` | Upload returns `attachment://` URL instead of signed URL |
| `apps/web/src/app/api/notes/[id]/route.ts` | GET resolves `attachment://` URLs; PATCH migrates legacy signed URLs |
| `apps/mobile/app/notes/[id].tsx` | Async URL resolution before editor init; migrate on save |

## Test plan

- [x] Unit tests for `attachment-url.ts` helpers (round-trip, edge cases, signed URL detection)
- [x] Unit tests for `resolve-urls.ts` (mock resolver, immutability, deduplication, nested content, migration)
- [x] Integration test updated: upload returns `attachment://` URL
- [x] Unit test: GET resolves `attachment://` URLs to signed URLs
- [x] Unit test: PATCH migrates legacy signed URLs to `attachment://`
- [x] All 96 shared tests pass, all 537 web tests pass
- [x] Typecheck clean across all packages
- [ ] E2E: upload image on web → reopen → image still displays

Closes #147

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added lazy-loading of attachment URLs with visual loading indicators on mobile
  * Enhanced file uploads to use standardized attachment URL format across platforms

* **Bug Fixes**
  * Improved content parsing robustness with fallback handling for edge cases
  * Strengthened error handling for file operations

* **Tests**
  * Expanded test coverage for URL resolution, format conversion, and content handling workflows

### CODE REVIEW

File: packages/shared/src/editor/attachment-url.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const SIGNED_URL_PATTERN = /\/storage\/v1\/object\/sign\/attachments\/([^?]+)/;
+
+export function isSignedStorageUrl(url: string): boolean {
+  return SIGNED_URL_PATTERN.test(url);
+}
+
+export function extractFilePathFromSignedUrl(url: string): string | null {
+  const match = url.match(SIGNED_URL_PATTERN);
+  if (!match) return null;
+  return decodeURIComponent(match[1]);
```
[the remark this answers is not in the document]
Comment (JakubAnderwald) [reply packages/shared/src/editor/attachment-url.ts]: Fixed: wrapped `decodeURIComponent` in a try/catch, returning `null` on decode failure so malformed URLs are skipped instead of crashing migration.
